### DISCUSSION TITLE
CardForm: Getting installments fails for values under 1000

### DISCUSSION THREAD

**Discussion by cuartas15** [mercadopago/sdk-js#198] [Q&A]
I'm using the latest version of mercadopago.js (v0.0.3)
Let's say I'm gonna charge for something that costs 500USD or EUR, when I type one of the testing Card numbers I get an error on console saying:

v2:2 MercadoPago.js - Failed to get installments. Use cardForm callbacks to intercept the error  TypeError: Cannot destructure property 'payer_costs' of 'e[0]' as it is undefined.
    at i (v2:2:73599)
    at _e.value (v2:2:66727)

I found out it's due to the amount set to cardform, if it's a number lower than 4 digits it'll fail and since there doesn't seem to be a way to specify the currency of the amount charged either (which I think it's necessary for this and the PHP SDK isn't it?) maybe it's hardcoded to expect latin american usual values that tend to go higher than 3 digits.

This is my code

`
async GetMercadoPagoObject() {
    await loadMercadoPago();

    this.mercadoPagoObject = new window.MercadoPago('MY PUBLIC KEY', {
        locale: this.mercadoPagoLanguages[this.currentLanguage] || 'en-US'
    });

    console.log('' + this.GetLocalStringToNumber(this.formedPrice));

    this.cardFormObject = this.mercadoPagoObject.cardForm({
        amount: '900',
        iframe: true,
        form: {
            id: 'credit-card-checkout',
            cardNumber: {
                id: 'cardNumber',
                placeholder: '',
            },
            expirationDate: {
                id: 'cardExpirationDate',
                placeholder: 'MM/YY',
            },
            securityCode: {
                id: 'cardSecurityCode',
                placeholder: '',
            },
            cardholderName: {
                id: 'cardHolderName',
                placeholder: '',
            },
            issuer: {
                id: 'cardIssuer',
                placeholder: '',
            },
            installments: {
                id: 'cardInstallments',
                placeholder: '',
            },
            identificationType: {
                id: 'cardIdentificationType',
                placeholder: '',
            },
            identificationNumber: {
                id: 'cardIdentificationNumber',
                placeholder: '',
            },
            cardholderEmail: {
                id: 'cardHolderEmail',
                placeholder: 'E-mail',
            }
        },
        callbacks: {
            onFormMounted: error => {
                if (error) return console.warn('Form Mounted handling error: ', error);

                selfComponent.$validator.reset();
            }
        }
    });
}
`

Any help is greatly appreciated

**Comment by lucmkz** [comment]
Hello @cuartas15, in order to be able to give you more adequate support, I need to understand a little more about what you are trying to do.

how are you setting which currency you're using? 
Your public key is from a user in which country? 
Tell me which test card you are using for testing.

This error seems to me that our API, which returns the installments, isn't returning any values. 

Thanks

**Comment by cuartas15** [comment]
> Hello @cuartas15, in order to be able to give you more adequate support, I need to understand a little more about what you are trying to do.
> 
> how are you setting which currency you're using? Your public key is from a user in which country? Tell me which test card you are using for testing.
> 
> This error seems to me that our API, which returns the installments, isn't returning any values.
> 
> Thanks

Hi, just to report, we figured out that Mercadopago SDK doesn't allow inputting the currency manually even if the request structure may have a 'currency_id' property, it doesn't do anything with that, it'll only charge with the currency the public and private key are attached to the country of the account created in mercadopago.
As to the question of how do we get the currency to charge anything for? that's something that we internally do in our software, not related to mercadopago

Thanks anyways
